### PR TITLE
1.7.1 j5int 1 fix virus download exception

### PR DESCRIPTION
**Pull request by ghost**
### Platforms affected
Windows

### Motivation and Context
- We run into a problem where our app on windows would login and then suddenly be stuck. Turned out we had a test virus file that caused the download to be stuck since the antivirus program blocks the file access.
- Also in this case it would not return the actual error but just return `null` as the error message. 

### Description

1. Add the actual error to the error callback when it is an unknown error so we can display the error message/log it on the client app
2. Ensure we sue the windows js `done` method that allows to handle any exception with a callback. Return the actual error to the client error callback if that happens

### Testing

Existing tests pass with no errors:

image

Tested it by changing an attachment id in the database for a standing order and a logbook attachment. I also added a virus file to each of the records. Also add a virus and change the attachment id for an industraForm

Login to mobile

Debug the code to go through the execution path:

devenv_HN8VuwsMvz

Changed the condition to ensure it hits the code path:

image

The errors are visible in the dialog when retry the sync.

image

Note: IndustraForm errors are not shown in this dialog only oData

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

File: src/windows/FileTransferProxy.js [hunk 1]
```diff
@@ -108,13 +108,15 @@ function doUpload (upload, uploadId, filePath, server, successCallback, errorCal
                     // in the other way, try to get response property
                     var response = upload.getResponseInformation();
                     if (!response) {
-                        resolve(new FTErr(FTErr.CONNECTION_ERR, source, server));
+                        resolve(new FTErr(FTErr.CONNECTION_ERR, source, server, null, null, error));
                     } else {
                         var reader = new Windows.Storage.Streams.DataReader(upload.getResultStreamAt(0));
-                        reader.loadAsync(upload.progress.bytesReceived).then(function (size) {
+                        reader.loadAsync(upload.progress.bytesReceived).done(function (size) {
```
Comment (yaseen-mowzer-octave) [inline src/windows/FileTransferProxy.js]: Where does the `done` method come from. I can't seem to find any documentation on it?

Comment (ghost) [reply src/windows/FileTransferProxy.js]: The docs are here: https://docs.microsoft.com/en-us/previous-versions/windows/apps/hh701079(v=win.10) . But I actually saw you can also use the second parameter for then which I didn't know. Let me use that instead.

Comment (ghost) [reply src/windows/FileTransferProxy.js]: https://docs.microsoft.com/en-us/previous-versions/windows/apps/hh700339(v=win.10)

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cordova-plugin-file-transfer",
-  "version": "1.7.1",
+  "version": "1.7.1-j5.1",
```
Comment (yaseen-mowzer-octave) [inline package.json:3]: This version is also in package-lock.json. I think maybe package-lock.json should be regenerated?

File: src/windows/FileTransferProxy.js (hunk 1 above)
Comment (ghost) [reply src/windows/FileTransferProxy.js]: I actually see that they use `done` in line 478 including the error handling. I have tried that `then` before when I was doing the server settings file and no success but it worked with `done`.

File: package.json (hunk 2 above)
Comment (ghost) [reply package.json:3]: There is no package-lock.json in this project

Comment (yaseen-mowzer-octave) [reply package.json:3]: Oh weird. It's only one the master branch. Ignore this then.

File: src/windows/FileTransferProxy.js (hunk 1 above)
Comment (ghost) [reply src/windows/FileTransferProxy.js]: ok changed it back to then. I checked the code and it has a second argument for onError

**Pull request merged**